mempool/src/core_mempool/transaction_store.rs

```diff
@@ -281,6 +282,9 @@ impl TransactionStore {
     pub(crate) fn get_required_balance(&mut self, address: &AccountAddress) -> u64 {
         self.transactions.get_mut(&address).map_or(0, |txns| {
             txns.iter().fold(0, |acc, (_, txn)| {
+                assume!(txn.gas_amount < u32::max_value() as u64); // seems more than reasonable
```

Comment:
Can this use `MAXIMUM_NUMBER_OF_GAS_UNITS` instead of `u32::max_value()`?

Also can this be bounded below by `MIN_TRANSACTION_GAS_UNITS` as well?

Reply:
That would be an improvement. Unfortunately these constants are defined as static variables in the VM crate, which is not referenced by this crate. Introducing such a reference seems intrusive and refactoring things to avoid it seems beyond the scope of this PR.